### PR TITLE
Week 9 - Update DG

### PR DESCRIPTION
**Pull request by ChongCheeKaiClarence**

### CODE REVIEW

**Review by florentianayuwono — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -234,6 +234,50 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 _{more aspects and alternatives to be added}_
 
+### Tag adding feature
+
+### Implementation
+
+The tag adding mechanism is facilitated bt `TagAddCommand` and `TagAddCommandParser`.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]: `bt`?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -234,6 +234,50 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+Additionally, The mechanism utilises the following operations in `UniqueTagList` and `UniquePersonList`.
+
+* `boolean hasTag(Tag tag)` - Checks if the tagList has the tag.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]: Both classes contains this method? If not, specify the class with #

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -234,6 +234,50 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+* `void setPerson(Person target, Person editedPerson)` - Sets the same person with the new tag.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]: Ditto

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -234,6 +234,50 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+* 
+
+These operations are exposed in the `Model` interface under the same method name.
+
+Given below is an example usage scenario and how the sorting mechanism behaves at each step.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]: `sorting`? Your copy-paste game quite weak, hahaha

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -234,6 +234,50 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+Step 5. The `tag add` command creates the same person with the new tag included, containing it in `editedPerson = createEditedPerson(Person personToEdit, Tag tag)`.
+
+Step 6. The `tag add` command replaces the old person with the new, updated person, calling `model#setPerson(Person target, Person editedPerson)`.
+
+The following activity diagram summarizes what happens when a user executes a tag add command:
+
+(insert activity diagram here)
+
+#### Design considerations:
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]: `Design consideration`, remove the `s:` so that this section looks nicer in the TOC

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -234,6 +234,50 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+
+**Aspect: How to implement tag add:**
+
+* **Alternative 1 (current choice):** Creates a new Person with the tag included.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]:
```suggestion
* **Alternative 1 (current choice):** Creates a new `Person` with the tag included.
```

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -234,6 +234,50 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+    * Pros: Prevents direct access into a Person's tags.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]:
```suggestion
    * Pros: Prevents direct access into the tags of a `Person`.
```

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -234,6 +234,50 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+    * Cons: Potential error occurs if some form of duplication is allowed.
+
+* **Alternative 2:** Directly add the tag into the Person .
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]:
```suggestion
* **Alternative 2:** Directly add the tag into the `Person` .
```

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -234,6 +234,50 @@ The following activity diagram summarizes what happens when a user executes a ne
...
+    * Pros: Easy to implement.
+    * Cons: Easy to access into a Person's tags. Could cause accidental bugs.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]:
```suggestion
    * Cons: Easy to access into the tags of a `Person`. Could cause accidental bugs.
```

**Review by Eugene-Ong-W-X — COMMENTED** [review]
Some suggestions you can consider.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (ChongCheeKaiClarence) [reply docs/DeveloperGuide.md]: Hmm. Looks like my commit from my laptop didn't get through. I'll make the changes.

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (ChongCheeKaiClarence) [reply docs/DeveloperGuide.md]: Oops. I was using yours as a reference.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (ChongCheeKaiClarence) [reply docs/DeveloperGuide.md]: Oh. So that's how it works. Alright. I'll make the change

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (ChongCheeKaiClarence) [reply docs/DeveloperGuide.md]: Done!

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (ChongCheeKaiClarence) [reply docs/DeveloperGuide.md]: I see. I have made the changes

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (ChongCheeKaiClarence) [reply docs/DeveloperGuide.md]: Changes done!

File: docs/DeveloperGuide.md (hunk 8 above)
Comment (ChongCheeKaiClarence) [reply docs/DeveloperGuide.md]: Changes made.

File: docs/DeveloperGuide.md (hunk 9 above)
Comment (ChongCheeKaiClarence) [reply docs/DeveloperGuide.md]: I made the changes.

File: docs/DeveloperGuide.md (hunk 7 above)
Comment (ChongCheeKaiClarence) [reply docs/DeveloperGuide.md]: Done!

**Comment by Eugene-Ong-W-X** [comment]
LGTM

**Pull request merged**